regolith/helpers/a_proposalhelper.py

```diff
@@ -0,0 +1,190 @@
+                       )
+    subpi.add_argument("-n", "--notes",
+                       help="Anything to note", default = ''
+                       )
+
+    #subpi.add_argument("-t", "--target_collection",
+    #                   help="Collection to add proposal to", default='proposals'
+    #                   )
+
+    return subpi
+
+class ProposalAdderHelper(DbHelperBase):
+    """Helper for adding a proposal to the proposals collection.
+
+       A proposal is a dictionary object describing a research or
+        project proposal submitted by the group.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "a_proposal"
+    needed_dbs = [f'{TARGET_COLL}', 'people', 'groups']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        #rc.coll = f"{rc.target_collection}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        gtx = self.gtx
+        rc = self.rc
+        now = dt.date.today()
+        key = f"{str(now.year)[2:]}_{''.join(rc.name.casefold().split()).strip()}"
+
+        coll = self.gtx[rc.coll]
+        pdocl = list(filter(lambda doc: doc["_id"] == key, coll))
+        if len(pdocl) > 0:
+            raise RuntimeError(
+                "This entry appears to already exist in the collection")
+        else:
+            pdoc = {}
+        pdoc.update({'_id': key,
+                     'amount': float(rc.amount)
+                     })
+        if isinstance(rc.authors, str):
+            pdoc.update({'authors': [rc.authors]})
+        else:
+            pdoc.update({'authors': rc.authors})
+        if rc.begin_date != 'tbd':
+            pdoc.update({'begin_date': date_parser.parse(rc.begin_date).date()})
+        else:
+            pdoc.update({'begin_date': rc.begin_date})
+        cpp_info = {}
+        if rc.cppflag:
+            cpp_info['cppflag'] = True
+        else:
+            cpp_info['cppflag'] = False
+        cpp_info['other_agencies_submitted'] = rc.other_agencies
+        cpp_info['institution'] = rc.institution
+        cpp_info['person_months_academic'] = int(rc.months_academic)
+        cpp_info['person_months_summer'] =  int(rc.months_summer)
+        cpp_info['project_scope'] = rc.scope
+        pdoc.update({'cpp_info':cpp_info})
+        pdoc.update({'currency': rc.currency})
+        if rc.due_date != 'tbd':
+            pdoc.update({'due_date': date_parser.parse(rc.due_date).date()})
+        else:
+            pdoc.update({'due_date': rc.due_date})
```

Comment:
we need to make sure the schema allows string or datetime (I think it does, but make sure it works)

Reply:
The schema as of now specifies string. Should I update it? Further, how do we input a `datetime` object in the CLI?

Reply:
Thanks Adiba, yes, please add ``date`` (rather than datetime here I think) and make it ``anyof_type`` to the schema (unrelated, but please can you look at this in the meetings schema also, I think it needs fixing there too).  The CLI always takes a string so we have to convert it in the code.  Please look at one of the other converters to see how to do it.

Reply:
`meetings` doesn't seem to have a schema in schemas.py, but there is a `due_date` input in the schema for `proposalReviews` which currently only allows string inputs, should I fix this to `"anyof_type": ["string", "date"]` as well?
Also, on the conversion, I am a little unsure what additional conversion you'd like me to add. I already have the functionality to convert a string of type `"YYYY-MM-DD"` to a `datetime` object. Should I also add the functionality to convert a string `"(YYYY, MM, DD)"` or `"datetime.date(YYYY, MM, DD)"` to a datetime object?

Reply:
yes please!  Its always good to clean code when you find/touch it, as long as it doesn't break anything.  

We will need a schema for meetings now it is fairly stable, but that would be for a different issue and a different PR.  You could help by posting that as a new issue on regolith and someone can volunteer for it in the future.  

For the date formatting, No. for the CLI we will enforce dates in the YYYY-MM-DD form.  If there is a gui somewhere in the future it can handle all the different ways people do it.

Reply:
done!